cypress/support/e2e.js

```diff
@@ -0,0 +1,48 @@
+import './commands.js'
+
+/**
+ * Custom login command, can be used to login or switch between sessions.
+ * Will cache and restore cookies, localStorage, and sessionStorage. See:
+ * https://docs.cypress.io/api/commands/session
+ */
+Cypress.Commands.add('login', (user) => {
+    cy.session(
+        user,
+        () => {
+            // Login via API
+            cy.request({
+                url: `${user.server}/dhis-web-commons-security/login.action`,
+                method: 'POST',
+                form: true,
+                followRedirect: true,
+                body: {
+                    j_username: user.name,
+                    j_password: user.password,
+                    '2fa_code': '',
+                },
+            })
+
+            // Set base url for the app platform
+            window.localStorage.setItem('DHIS2_BASE_URL', user.server)
+        },
+        {
+            validate: () => {
+                // Check API is returning the expected response
+                cy.request(`${user.server}/api/me`).should((response) => {
+                    expect(response.status).to.eq(200)
+                    expect(response.body.username).to.eq(user.name)
+                })
+            },
+            cacheAcrossSpecs: false,
+        }
+    )
+})
+
+// Log in before each test, if not already logged in
+beforeEach(() => {
+    cy.login({
+        name: Cypress.env('dhis2Username'),
+        password: Cypress.env('dhis2Password'),
+        server: Cypress.env('dhis2BaseUrl'),
+    })
+})
```

Comment:
This part of the PR unsettles me slightly. In `@dhis2/cypress-commands` we have a function called `enableAutlogin` which does something very similar to this, but when switching to use that the test suite became flaky.

Since `enableAutoLogin` sets `cacheAcrossSpecs` to `true` and the flakiness I was seeing was usually to do with tests being stuck at the login modal, the difference between `enableAutoLogin` and the login command here could explain things.

However, `@dhis2/cypress-commands` also exports a [`loginByApi`](https://github.com/dhis2/cli-utils-cypress/blob/master/packages/cypress-commands/src/commands/loginByApi.js) and a [`validateUserIsLoggedIn`](https://github.com/dhis2/cli-utils-cypress/blob/master/packages/cypress-commands/src/commands/validateUserIsLoggedIn.js) command. These commands do exactly the same thing as what I have above. So using this custom code or the commands from our lib _should_ produce the same results. But they didn't, and this both puzzles and concerns me.